sdk/test-utils/recorder/src/utils/requestBodyTransform.ts

```diff
@@ -14,35 +14,38 @@ export type RequestBodyTransformsType = {
 /**
  * Provides the default RequestBodyTransforms that need to be applied on the generated recordings
  */
-export const defaultRequestBodyTransforms: RequestBodyTransformsType = {
+export const defaultRequestBodyTransforms: Required<RequestBodyTransformsType> = {
   stringTransforms: [
-    // Identity v2 with the new msal, has unique request-ids in request body, to be able to
-    // match the ids in playback we apply the following method on the request body
-    // 1. before saving the recording
-    //    and
-    // 2. as a filter on the new requests to be able to match the request bodies
+    // 1. Identity v2 with the new msal, has unique request-ids in request body, to be able to
+    //    match the ids in playback we apply the following method on the request body
+    //    - before saving the recording
+    //      and
+    //    - as a filter on the new requests to be able to match the request bodies
+    // 2. Sanitizes the scope values in the recordings - to reduce the noise from cred scan reports
     (body: string) =>
-      body.replace(/client-request-id=[^&]*/g, "client-request-id=client-request-id")
+      body
+        .replace(/client-request-id=[^&"]*/g, "client-request-id=client-request-id")
+        .replace(/scope=https%3A%2F%2F[^&"]*/g, "scope=https%3A%2F%2Fsanitized%2F")
```

Comment:
maybe just `[^&]*`?  Although `"` appears in the recording files but I think it denotes a string value, it will not appear in the string itself, right?

Reply:
No, we're literally parsing it.
It'll look something like `"client-request-id=11111111-1111-1111-1111-111111111111&client_secret=azure_client_secret"`
with the " quotes.